services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/_core/_events_utils.py

```diff
@@ -442,3 +451,24 @@ async def prepare_services_environment(
     )
 
     scheduler_data.dynamic_sidecar.is_service_environment_ready = True
+
+
+async def get_allow_metrics_collection(
+    app: FastAPI, user_id: UserID, product_name: ProductName
+) -> bool:
+    repo = get_repository(app, UserPreferencesFrontendRepository)
+    preference: FrontendUserPreference | None = await repo.get_user_preference(
+        user_id=user_id,
+        product_name=product_name,
+        preference_class=AllowMetricsCollectionFrontendUserPreference,
+    )
+
+    if preference is None:
+        return cast(
+            bool, AllowMetricsCollectionFrontendUserPreference.get_default_value()
```

Comment:
can this actually be? in which case does the preference not exist then?

Reply:
why do we need to care about someone doing wrong stuff in the DB?
I find this a bit cumbersome if you need to check everytime for the existense of a preference. It could be defaulted (which it is already right?)